docs/walkthrough.rst

```diff
@@ -0,0 +1,199 @@
+error in this case.
+
+All from selected
+^^^^^^^^^^^^^^^^^
+The fourth option is to apply the scripts to all of the selected objects and the children  \
+objects. Selecting a Project and using the option **<all (from selected)>** with result in \
+applying the scripts to the selected project, to the linked datasets and the linked images \
+bellow. Note that this option does not exist for the import of annotation from csv.
+
+Exporting annotations
+---------------------
+
+Exporting annotations will generate a .csv file with the header names \
+expected by the import .csv script. This has two use:
+
+* export existing key-value pairs (along other things like namespace or tags).
+  This can be useful to modify them before re-import, or to transfer annotations
+  across groups.
+* export a list of objects with their name and ID.
+
+.. figure:: images/annotation_04_export_output.png
+
+   *Exported .csv with only the object ID and name*
+..
+
+  Tip: If you have Key-Value pairs attached to your objects that you do not \
+  wish to export (to create a template like shown above), specify a not-in-use \
+  namespace (any random input will work). This does not affect tags.
+
+
+Importing annotations
+---------------------
+
+Starting from the file we exported as explained in the previous section, \
+we proceed to edit it within a spreadsheet editor, adding more columns to the csv, \
+including a row specifying the namespace to assign for each key:
+
+.. figure:: images/annotation_05_KV_to_import.png
+
+   *Result of populating the csv shown in a text editor*
+..
+
+We added several columns to annotate our dataset with Key-Value pairs \
+following the `REMBI <https://doi.org/10.1038/s41592-021-01166-8>`_ guidelines \
+(after saving the document, our ``TAB`` separator became ``,``).
+
+We proceed and start the script "*Import from CSV*".
+
+.. figure:: images/annotation_06_script_import.png
+
+   *Selection of all the datasets of a project and the Import from CSV script.*
+..
+
+There are many parameters here that could help you fine tune the way to import annotations. More \
+  on that in the :doc:`parameter description section</parameter_description>` of this guide.
+
+
+We can then see in the OMERO activities that the Key-Value pairs were added to 5 \
+datasets out of the 11 present in this project (as expected). We also see three set of \
+key-value pairs, one for each namespace we used in the csv.
+
+.. figure:: images/annotation_07_KV_import_printout.png
+
+  *The script output (5 entries in the csv matched to 5 dataset out of 11) and the\
+  resulting key-value pairs annotation.*
+..
+
+Converting the Key-Value pairs namespace
+----------------------------------------
+
+Key-Value pairs are assigned a category/label (known as namespace). \
+This grants flexibility so that different annotations can be \
+distinguished or isolated (like for exporting/deleting only those with a given \
```

Comment:
too many brackets - not helpful for understanding (I do it too sometimes... but not very good practice :0

Reply:
reformulated without brackets